src/lib.rs

```diff
@@ -1574,14 +1620,150 @@ mod test {
         assert!(!_NEG1_2.is_integer());
     }
 
+    #[cfg(not(feature = "std"))]
+    use core::fmt;
+    #[cfg(not(feature = "std"))]
+    use fmt::Write;
+    #[cfg(not(feature = "std"))]
+    const TESTER_BUF_SIZE: usize = 32;
+    #[cfg(not(feature = "std"))]
+    #[derive(Debug)]
+    struct NoStdTester {
+        cursor: usize,
+        buf: [u8; TESTER_BUF_SIZE],
+    }
+
+    #[cfg(not(feature = "std"))]
+    impl NoStdTester {
+        fn new() -> NoStdTester {
+            NoStdTester {
+                buf: [0; TESTER_BUF_SIZE],
+                cursor: 0,
+            }
+        }
+
+        fn clear(&mut self) {
+            self.cursor = 0;
+        }
+    }
+
+    #[cfg(not(feature = "std"))]
+    impl Write for NoStdTester {
+        fn write_str(&mut self, s: &str) -> fmt::Result {
+            for byte in s.bytes() {
+                self.buf[self.cursor] = byte;
+                self.cursor += 1;
+                if self.cursor >= self.buf.len() {
+                    return Err(fmt::Error {});
+                }
+            }
+            Ok(())
+        }
+    }
+
+    #[cfg(not(feature = "std"))]
+    impl PartialEq<str> for NoStdTester {
+        fn eq(&self, other: &str) -> bool {
+            let other = other.as_bytes();
+            for index in 0..self.cursor {
+                if self.buf.get(index) != other.get(index) {
+                    return false;
+                }
+            }
+            true
+        }
+    }
+
+    #[test]
+    #[cfg(not(feature = "std"))]
+    fn test_show() {
+        let err_msg = "Formatted output too long";
+        let mut tester = NoStdTester::new();
+        write!(tester, "{:o}", _8).expect(err_msg);
+        assert_eq!(tester, *"10");
```

Comment:
Can you please add an alt test, as well as non-integral values?

Reply:
My plan for this code was to create a macro that let me test in both `std` and `no_std` environments easily. i.e.
```rust
macro_rules! assert_fmt_eq {/*implementation goes here*/}

assert_fmt_eq!("2", format_args!("{}", _2));
```
which expands to use `NoStdTester` or normal `format!` based on the `std` feature

 I wanted to make sure `NoStdTester` idea (hack?) wasn't stupid first. Specifically that there is no easier way to do this.